next/13/replace-next-head/index.ts

```diff
@@ -1054,37 +1056,101 @@ export const buildComponentMetadata = (
 	};
 };
 
+const removeMdxNodes = (
+	{ visitMdxAst, filterMdxAst, stringifyMdx, parseMdx }: Dependencies,
+	content: string,
+): string => {
+	if (!filterMdxAst || !visitMdxAst || !stringifyMdx || !parseMdx) {
+		return content;
+	}
+
+	try {
+		const tree = parseMdx(content);
+
+		const newTree = filterMdxAst(tree, (node) => {
+			return ['root', 'mdxJsxFlowElement', 'mdxjsEsm'].includes(
+				node.type,
+			);
+		});
+
+		if (newTree === undefined) {
+			return content;
+		}
+
+		const mdxJsxFlowElements: string[] = [];
+
+		visitMdxAst(newTree, (node) => {
+			if (node.type === 'mdxJsxFlowElement') {
+				// @ts-expect-error stringify any node
+				const value = stringifyMdx(node);
+				mdxJsxFlowElements.push(value);
+			}
+		});
+
+		let stringifiedMdx = stringifyMdx(newTree);
+
+		mdxJsxFlowElements.forEach((value) => {
+			const r = new RegExp(`(?<!{)${value}(?!})`, 'gm');
+			stringifiedMdx = stringifiedMdx.replace(r, `{${value}}`);
+		});
+
+		return stringifiedMdx;
+	} catch (e) {
+		console.error(e);
+		return content;
+	}
+};
+
 const initTsMorphProject = async (
-	tsmorph: Dependencies['tsmorph'],
+	dependencies: Dependencies,
 	unifiedFileSystem: Dependencies['unifiedFileSystem'],
 	rootPath: string,
 	compilerOptions?: tsmorph.CompilerOptions,
 ) => {
+	const { tsmorph } = dependencies;
+
 	const _compilerOptions = {
 		...defaultCompilerOptions,
 		...compilerOptions,
 		baseUrl: rootPath,
 	};
 
-	project = new tsmorph.Project({
+	const project = new tsmorph.Project({
 		useInMemoryFileSystem: true,
 		skipFileDependencyResolution: true,
 		compilerOptions: _compilerOptions,
 	});
 
 	const allFilePaths = await unifiedFileSystem.getFilePaths(
 		rootPath,
-		['**/*.{jsx,tsx,ts,js,cjs,ejs}'],
+		['**/*.{jsx,tsx,ts,js,cjs,ejs,mdx}'],
```

Comment:
we will need to see if these files are included if the extension passes its own include glob patterns